### DISCUSSION TITLE
font() not working with quarto

### DISCUSSION THREAD

**Discussion by philimho** [davidgohel/flextable#624] [Q&A] [answered]
Hi, I am trying to change the font in a flextable-body in a pdf output with the 'font()' function. This works fine with a markdown-file:
``````
---
output:
  pdf_document:
    latex_engine: xelatex
    keep_tex: true
title: "Mrkdwn_test"
mainfont: Roboto
---
## Lorem

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin suscipit, nulla eu commodo euismod, justo nisi scelerisque nulla, ut imperdiet turpis lorem vel dolor. 
```{r, echo = FALSE}
library(flextable)
flextable::set_flextable_defaults(
    font.family = "Roboto", 
  )
ft <- flextable::flextable(iris) |>
  flextable::font(fontname = "Roboto Condensed", part = "body")

ft
```
``````

The resulting pdf uses Roboto correctly for the standalone text and the column headers, while using Roboto Condensed in the flextable-body.
![image](https://github.com/davidgohel/flextable/assets/166725450/231f2719-1a2d-40e5-82bd-d79aa7f9caf9)

Now trying to do the same in quarto:
``````
---
title: "quarto_test"
format:
  pdf:
    documentclass: article
pdf-engine: xelatex
mainfont: Roboto
keep-tex: true
---

## Lorem

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin suscipit, nulla eu commodo euismod, justo nisi scelerisque nulla, ut imperdiet turpis lorem vel dolor. Proin eu tincidunt nisi. Mauris volutpat purus ut iaculis venenatis. Sed a purus arcu. Maecenas non nibh libero. Ut gravida velit augue, a rhoncus massa lobortis quis. Vestibulum vestibulum dolor ac iaculis faucibus.

```{r, echo = FALSE}
library(flextable)
flextable::set_flextable_defaults(
    font.family = "Roboto", 
  )
ft <- flextable::flextable(iris) |>
  flextable::font(fontname = "Roboto Condensed", part = "body")

ft
```
``````
Results in a pdf-file with only the mainfont, ignoring the defined font for the body of the flextable:
![image](https://github.com/davidgohel/flextable/assets/166725450/927d4265-f98b-433a-bb1c-0f4be49cb48e)

Any idea why the font() is not applied in the .tex file created by quarto? Sorry if this is a quarto problem rather than a flextable issue.

**Comment by davidgohel** [comment]
I don't see a difference between the first and the second image.

Could you add the result of `gdtools::font_family_exists("Roboto Condensed")` and `gdtools::font_family_exists("Roboto")`?


<img width="468" alt="Capture d’écran 2024-04-12 à 14 33 09" src="https://github.com/davidgohel/flextable/assets/4331618/cad19b1f-91e1-405e-897c-188d8164738f">

``````
---
title: "quarto_test"
format:
  pdf:
    documentclass: article
pdf-engine: xelatex
mainfont: Roboto
keep-tex: true
---

## Lorem

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin suscipit, nulla eu commodo euismod, justo nisi scelerisque nulla, ut imperdiet turpis lorem vel dolor. Proin eu tincidunt nisi. Mauris volutpat purus ut iaculis venenatis. Sed a purus arcu. Maecenas non nibh libero. Ut gravida velit augue, a rhoncus massa lobortis quis. Vestibulum vestibulum dolor ac iaculis faucibus.

```{r, echo = FALSE}
library(flextable)
flextable::set_flextable_defaults(
    font.family = "Roboto", 
    table.layout = "autofit"
  )
ft <- flextable::flextable(head(iris)) |>
  flextable::font(fontname = "Roboto Condensed", part = "body")

ft
```

``````


On my machine, I installed both

**Reply by philimho** [reply]
Both font_family_exists commands return TRUE.

To make the problem more visibale i swapped "Roboto Condensed" with a different font, "Univers":
Quarto only uses Roboto (main font):
![image](https://github.com/davidgohel/flextable/assets/166725450/588b617d-b344-4480-a520-aab7c7965165)

Markdown correctly uses Univers in the flextable-body:
![image](https://github.com/davidgohel/flextable/assets/166725450/dacc4faa-8ed5-44d4-95af-a2c4c058269e)

Looking at the two .tex-files, the one generated with quarto seems to be missing the ```\global\setmainfont{Univers}``` commands in each cell:
Quarto:
![image](https://github.com/davidgohel/flextable/assets/166725450/8656bd0b-9925-47e5-80f2-5c1883006078)

Markdown:
![image](https://github.com/davidgohel/flextable/assets/166725450/83eaec80-906d-4419-b34c-7886b6033291)

**Comment by davidgohel** [comment]
OK, I'll have a look.

**Reply by davidgohel** [reply, marked as answer]
that was a bug in flextable, it should be fixed now

**Reply by philimho** [reply]
Thank you for the fast fix
